### ISSUE TITLE
Identity providers - NIP-05 extension idea

### ISSUE THREAD

**Issue by ondra-novak** [nostr-protocol/nips#603]
Hello,

I would like to propose an idea that could potentially be considered for a new NIP. This idea builds upon NIP-05.

If you have a profile on a DNS/server, it is likely that you either have control over the DNS/server or trust the provider who allows you to be verified on that server.

What if we enable providers to offer "authentication" through this server, similar to how identity providers work with OpenID?

Here's how it could work:

- The identity provider would store your private key.
- You would provide proof of your identity to the identity provider (the method is not specified, for example, by entering a password).
- The identity provider can then send the private key to an application that requests this operation.
- Alternatively, the identity provider could provide a signing tool to sign events. This open possibility to define scopes as are known in OAuth.

I would like to gather the opinions of other contributors regarding this idea.

**Comment by fiatjaf** [comment]
You could have signing providers that work using https://github.com/nostr-protocol/nips/blob/master/46.md, that doesn't require any relationship with NIP-05.

**Comment by ondra-novak** [comment]
I don't think we quite understand each other. NIP-05 introduces identification, which is very similar to the identification of the user to his e-mail box. In the same sense, a regular user will want to manage their account.

Common users are far from understanding how to deal with keys. They need an expert to arrange it for them. What would help them is if logging into applications was implemented in the same way as logging into e-mail. Enter your email and your password. And done. If they lose their password, it is then the administrator's responsibility to somehow help them regain access to the account.

This is not against the point of decentralization, as it still allows for community account managers. And also the possibility to move the account from one administrator to another.

My proposal would eventually extend the .well-known section with a URL to an API call, which would arrange OAuth login and then the operation with the stored private key. This call to standardize and allow applications classic login.

**Comment by arthurfranca** [comment]
Yeah it would make login easy for non-tech people.
I got what you want and could draft a NIP but I won't dare opening more PRs. I have like 10 open PRs hanging in limbo lol 😅️

**Comment by ondra-novak** [comment]
I am planning to try to implement a proof of concept on a relay (see my profile https://github.com/ondra-novak/nostr_server) once the project will be in adequate condition. I want to see, which approach is better.

It could be implemented as REST api, or it can extend signer tool (nip 46) with some new commands - for example command oauth_connect(), which returns URL to web login form.

Problem with NIP 46 is that this feature is designed for applications, not for web services (many operations "shows popup", which is impossible when the signer is on a server).
